onnxruntime/core/common/logging/logging.cc

```diff
@@ -276,5 +281,79 @@ Severity OverrideLevelWithEtw(Severity originalSeverity) {
   return originalSeverity;
 }
 
+#ifdef _WIN32
+void LoggingManager::AddEtwSink(logging::Severity etwSeverity) {
```

Comment:
Can we restructure a little? Seems like there's a fairly generic add/remove of a new Sink to a composite Sink (which may need to be created) that's a little entangled with some ETW specific logic. 

We could potentially decouple by 
- passing in `std::function<std::unique_ptr<ISink>())` to abstract the new sink type and the construction requirements of that
- passing in SinkType to add/remove to enforce not adding a sink of that type if it already exists, and to do the matching to remove
- moving the ETW specific check on the registration manager being enabled to the calling code. 
  - as that code knows it wants to add an ETW sink it's ETW specific already